requests/adapters.py

```diff
@@ -272,13 +272,10 @@ def request_url(self, request, proxies):
         :class:`HTTPAdapter <requests.adapters.HTTPAdapter>`.
 
         :param request: The :class:`PreparedRequest <PreparedRequest>` being sent.
-        :param proxies: A dictionary of schemes to proxy URLs.
+        :param proxies: A dictionary of schemes or schemes and hosts to proxy URLs.
         """
-        proxies = proxies or {}
-        scheme = urlparse(request.url).scheme
-        proxy = proxies.get(scheme)
-
-        if proxy and scheme != 'https':
+        proxy = select_proxy(request.url, proxies)
+        if proxy and not request.url.lower().startswith('https'):
```

Comment:
I used that because it is also used at https://github.com/jasongrout/requests/blob/per-host-proxy/requests/adapters.py#L171.  I wondered about the correctness.  Sounds like we should fix it there too?